doc/api/n-api.md

```diff
@@ -0,0 +1,3150 @@
+information about the error.
+
+##### Return value
+- `napi_ok` if the API succeeded.
+
+##### Description
+This API retrieves a `napi_extended_error_info` structure with information
+about the last error tha occured.
+
+
+### Exceptions
+
+Any N-API function may result in a pending JavaScript exception. This is
+obviously the case for any function that may cause the execution of JavaScript
+but N-API specifies that an exception may be pending on return from any
+of the API functions.
+
+If the `napi_status` returned by a function is `napi_ok` then no
+exception is pending and no additional action is required. If the
+`napi_status` returned is anything other than `napi_ok` or
+`napi_pending_exception`, and you want to try to recover and continue
+instead of simply returning immediately, then you must call the
+following function in order to determine if an exception is pending or not
+and then check the value of `result`:
+
+```C
+napi_is_exception_pending()
+```
+
+When an exception is pending you will generally follow one of two approaches.
+
+The first appoach is to do any appropriate cleanup and then return so that
+execution will return to JavaScript. As part of the transition back to
+JavaScript the exception will be thrown at the point in the JavaScript
+code where the native method was invoked. The behavior of most N-API calls
+is unspecified while an exception is pending, and many will simply return
+`napi_pending_exception`, so it is important to do as little as possible
+and then return to JavaScript where the exception can be handled.
+
+The second approach is to try to handle the exception. There will be cases
+where the native code can catch the exception, take the appropriate action,
+and then continue. It is recommended that you only do this in specific cases
+where you know you can safely handle the exception. In these cases you can
+use the following function to get and clear the exception:
+
+```C
+napi_get_and_clear_last_exception()
+```
+
+On success, result will contain the handle to the last JavaScript Object
+thrown. If you determine after retrieving the exception that you cannot
+handle it after all, you can re-throw it with the following function:
+
+```C
+napi_throw()
+```
+
+where error is the JavaScript Error object to be thrown.
+
+The following utility functions are also available in case native code
+needs to throw an exception or determine if a `napi_value` is an instance
+of a JavaScript `Error` object.
+
+```C
+napi_throw_error()
+napi_throw_type_error()
+napi_throw_range_error()
+napi_status napi_is_error()
+```
+
+The following utility functions are also available in case native
+code needs to create an Error object:
+
+```
+napi_create_error()
+napi_create_type_error()
+napi_create_range_error()
+```
+
+where result is the napi_value that refers to the newly created
+JavaScript Error object.
+
+#### napi_throw
+##### Signature
+```C
+NODE_EXTERN napi_status napi_throw(napi_env e, napi_value error);
```

Comment:
Here and throughout, can we make sure that the signature has the same names as the listed parameters?

e.g. `e` should be `env`  to match.